pagerduty/user.go

```diff
@@ -383,14 +424,52 @@ func (s *UserService) DeleteContactMethod(userID, contactMethodID string) (*Resp
 	return resp, err
 }
 
+// ListContactMethods lists contact methods for a user.
+func (s *UserService) ListNotificationRules(userID string) (*ListNotificationRulesResponse, *Response, error) {
+	u := fmt.Sprintf("/users/%s/notification_rules", userID)
+	v := new(ListNotificationRulesResponse)
+
+	resp, err := s.client.newRequestDo("GET", u, nil, nil, &v)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return v, resp, nil
+}
+
 // CreateNotificationRule creates a new notification rule for a user.
 func (s *UserService) CreateNotificationRule(userID string, rule *NotificationRule) (*NotificationRule, *Response, error) {
 	u := fmt.Sprintf("/users/%s/notification_rules", userID)
 	v := new(NotificationRulePayload)
 
 	resp, err := s.client.newRequestDo("POST", u, nil, &NotificationRulePayload{NotificationRule: rule}, &v)
 	if err != nil {
-		return nil, nil, err
+		if e, ok := err.(*Error); ok && strings.Compare(fmt.Sprintf("%v", e.Errors), "[Channel Start delay must be unique for a given contact method]") == 0 {
+			lResp, _, lErr := s.ListNotificationRules(userID)
+			if lErr != nil {
+				return nil, nil, fmt.Errorf("[Channel Start delay must be unique for a given contact method]. Failed to fetch existing rules: %w", lErr)
+			}
+
+			foundRule := false
+			for _, nr := range lResp.NotificationRules {
+				if isSameNotificationRule(nr, rule) {
+					gRule, gResp, gErr := s.GetNotificationRule(userID, nr.ID)
+					if gErr != nil {
+						return nil, nil, gErr
+					}
+					v.NotificationRule = gRule
+					resp = gResp
+					foundRule = true
+					break
+				}
+			}
+
+			if !foundRule {
+				return nil, nil, err
+			}
+		} else {
```

Comment:
Can we move this up? I think this is too far from the original condition and too many unnecessary tabs to keep track of.